src/main/java/com/google/sps/data/NSWGovAdaptor.java

```diff
@@ -0,0 +1,126 @@
+package com.google.sps.data;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+/** File to run the GET request in java before putting it in the main adaptor function. 
+ */
+public class NSWGovAdaptor {
+
+  private HashMap<Integer, GovSiteDetails> dataMap;
+
+  public NSWGovAdaptor() throws Exception {
+    dataMap = new HashMap<>();
+    this.updateSiteInfo();
+  }
+
+  public HashMap<Integer, GovSiteDetails> getMap(HashMap<Integer, GovSiteDetails> dataMap) {
+    return this.dataMap;
```

Comment:
dataMap can be accessed within functions of this class, why do you need this function? Recommend some reading about encapsulation, https://www.tutorialspoint.com/java/java_encapsulation.htm and re-consider this .

Reply:
I need this function as my TestAdaptor.java needs to access the map to print out data to the terminal so I can see if I am returning the right things -  although yes, we wouldn't need it in the production code